docs/book/developer-guide/steps-pipelines/runtime-configuration.md

```diff
@@ -0,0 +1,333 @@
+```
+
+The `path_to_config.yaml` needs to have the following structure:
+
+```yaml
+steps:
+  <STEP_NAME_IN_PIPELINE>:
+    parameters:
+      <PARAMETER_NAME>: <PARAMETER_VALUE>
+      ...
+    ...
+```
+
+For our example from above, we could use the following configuration file:
+
+```yaml
+steps:
+  step_2:
+    parameters:
+      gamma: 0.01
+```
+
+{% hint style="info" %}
+Note that `svc_trainer()` still has to be defined to have a
+`config: SVCTrainerStepConfig` argument. The difference here is only that we
+provide `gamma` via a config file before running the pipeline, instead of
+explicitly passing a `SVCTrainerStepConfig` object during the step creation.
+{% endhint %}
+
+### Configuring the entire pipeline at runtime
+
+For production settings, you might want to use config files not only for your
+parameters, but even for choosing what code gets executed.
+This way, you can define entire pipeline runs without changing the code.
+
+To run pipelines in this way, you can use the `zenml pipeline run` command with
+`-c` argument:
+
+```shell
+zenml pipeline run <PATH_TO_PIPELINE_PYTHON_FILE> -c <PATH_TO_CONFIG_YAML_FILE>
+```
+
+`<PATH_TO_PIPELINE_PYTHON_FILE>` should point to the Python file where your
+pipeline function or class is defined. Your steps can also be in that file,
+but they do not need to. If your steps are defined in separate code files, you
+can instead specify that in the YAML, as we will see [below](#defining-which-steps-to-use).
+
+{% hint style="warning" %}
+Do **not** instantiate and run your pipeline within the python file that you
+want to run using the CLI, else your pipeline will be run twice, possibly with 
+different configurations.
+{% endhint %}
+
+If you want to dynamically configure the entire pipeline, your config file will
+need a bit more information:
+- The name of the function or class of your pipeline in 
+`<PATH_TO_PIPELINE_PYTHON_FILE>`,
+- The name of the function or class of each step, optionally with additional
+path to the the code file where it is defined (if it is not in 
+`<PATH_TO_PIPELINE_PYTHON_FILE>`),
+- Optionally, the name of each materializer (about which you will learn
+later in the section on [Materializers](materializer.md)).
+
+Overall, the required structure of such a YAML should look like this:
+
+```yaml
+name: <PIPELINE_CLASS_OR_FUNCTION_NAME>
+steps:
+  <STEP_NAME_IN_PIPELINE>:
+    source:
+      name: <STEP_CLASS_OR_FUNCTION_NAME>
+      file: <PATH_TO_STEP_PYTHON_FILE>  # (optional)
+    parameters:  # (optional)
+      <PARAMETER_NAME>: <PARAMETER_VALUE>
+      <SOME_OTHER_PARAMETER>: ...
+    materializers:  # (optional)
+      <NAME_OF_OUTPUT_OF_STEP>:
+        name: <MATERIALIZER_CLASS_NAME>
+        file: <PATH_TO_MATERIALIZER_PYTHON_FILE>
+      <SOME_OTHER_OUTPUT>:
+        ...
+  <SOME_OTHER_STEP>:
+    ...
+```
+
+This might seem daunting at first, so let us go over it one by one:
+
+### Defining which pipeline to use
```

Comment:
Added "in YAML" at the end of both sub headings. 

And no, this should be ### since it is independent from the section on configuring only step parameters in YAML.